addons/account_edi/models/mail_thread.py

```diff
@@ -0,0 +1,10 @@
+from odoo import models
+
+
+class MailThread(models.AbstractModel):
+    _inherit = 'mail.thread'
+
+    def _message_set_main_attachment_id(self, attachment_ids):
+        if self.message_main_attachment_id and len(attachment_ids) > 1 and self.message_main_attachment_id.mimetype == 'application/xml':
```

Comment:
I don't know if it's possible, but maybe we could go from 1 attachment XML to 1 attachment PDF (so maybe we should just check attachment_ids)

also not sure if this is necessary, but if we wanted to target only the EDI xml for invoice, maybe we could do this code:

- on account.move model
- and not just always ignore application/xml but rather remove it if matches `self.edi_document_ids.attachment_id`
- maybe if it's never good to have EDI xml as main attachment, we should not remove it, but never set it (filter out the xml EDI attachments)

Reply:
I agree with Nicolas for this part.

Reply:
> on account.move model
and not just always ignore application/xml but rather remove it if matches self.edi_document_ids.attachment_id

This part sorry :)

Btw the other part is good too but I think it's more a functional choice than a fix :sweat_smile: